packages/hyperion-autologging/src/ALSurfaceContext.ts

```diff
@@ -17,37 +20,55 @@ export type InitOptions =
   }>;
 
 
-type ALSurfaceContextValue = Readonly<{
+type ALSurfaceContextValue<
+  DataType extends ALFlowletDataType,
+  FlowletType extends Flowlet<DataType>
+> = Readonly<{
   surface: string,
+  flowlet: FlowletType | null,
 }>;
 
-const DefaultSurfaceContext: ALSurfaceContextValue = {
-  surface: '',
+
+
+function getDefaultSurfaceContext<
+  DataType extends ALFlowletDataType,
+  FlowletType extends Flowlet<DataType>
+>(): ALSurfaceContextValue<DataType, FlowletType> {
+  return {
+    surface: '',
+    flowlet: null,
+  };
 };
 
-export let ALSurfaceContext: React.Context<ALSurfaceContextValue> | null = null;
+export let ALSurfaceContext: React.Context<any> | null = null;
```

Comment:
why 'any'? I assume that's because of the type parameters that you added, which tells you why it is not a good idea to do that and then break the type safety somewhere else.